gap/base/recognition.gi

```diff
@@ -771,6 +771,9 @@ InstallGlobalFunction( FindKernelDoNothing,
     return true;
   end );
 
+# multiplies some of the generators (goes through the list (of generators)
+# and multiplies an entry with probability 1/2 to the Subproduct of the
```

Comment:
Which entry? a random one? Or perhaps "each entry"?  Why is Subproduct in uppercase? Is this "subproduct" as in "subdirect product" or what?